python/ffsim/linalg/double_factorized_decomposition.py

```diff
@@ -520,6 +519,12 @@ def double_factorized_t2(
         coeffs[:, :, None, None] * eigs[:, :, :, None] * eigs[:, :, None, :]
     )
 
+    orbital_rotations = orbital_rotations.reshape(-1, norb, norb)
+    diag_coulomb_mats = diag_coulomb_mats.reshape(-1, norb, norb)
+    if max_vecs:
+        orbital_rotations = orbital_rotations[:max_vecs]
+        diag_coulomb_mats = diag_coulomb_mats[:max_vecs]
+
     return diag_coulomb_mats, orbital_rotations
```

Comment:
```suggestion
    return diag_coulomb_mats[:max_vecs], orbital_rotations[:max_vecs]
```
This works even if `max_vecs` is `None`.